tool/tsh/common/tsh.go

```diff
@@ -688,13 +689,56 @@ func initLogger(cf *CLIConf) {
 	}
 }
 
+// transformArgs performs OpenSSH transformation on the passed in CLI args.
+func transformArgs(args []string) ([]string, error) {
+	var cmdline []string
+
+	// If the executable name is "ssh" or "scp" transform args to "tsh ssh ..."
+	// or "tsh scp ...".
+	switch filepath.Base(os.Args[0]) {
+	case "ssh":
+		cmdline = append([]string{"ssh"}, args...)
+	case "scp":
+		cmdline = append([]string{"scp"}, args...)
+	default:
+		cmdline = args
+	}
+
+	// The following flags are OpenSSH compatibility flags. They are used for
+	// people that alias "ssh" to "tsh ssh." The following OpenSSH flags are
+	// implemented. From "man 1 ssh":
+	//
+	// * "-V Display the version number and exit."
+	// * "-T Disable pseudo-terminal allocation."
+	if slices.Contains(cmdline, "-V") {
+		modules.GetModules().PrintVersion()
+
+		return nil, &common.ExitCodeError{
+			Code: 0,
+		}
+	}
+	i := slices.Index(cmdline, "-T")
+	if i > 0 {
+		cmdline = slices.Replace(cmdline, i, i+1, []string{"--no-tty"}...)
+	}
```

Comment:
This won't handle short options grouped together in a single argument.

Kingpin's behavior already prohibits repeated short boolean flags AFAICT, so we could just have a secondary hidden flag for `T` and error out if both `t` and `T` are passed in (even though this would still technically be in violation of the OpenSSH cli api, since that follows "last option wins" instead).

Reply:
@espadolini That's how I originally implemented this feature. @zmb3 you were not a fan, what do you think?

* https://github.com/gravitational/teleport/commit/e891855ff89056185979f8a2ad9502ce864d0034
* https://github.com/gravitational/teleport/commit/10fa8951b51fae44ecf78a1a56f4822fe7853110

I don't have strong feelings either way.

Reply:
@espadolini @zmb3 I updated the implementation, please take another look.